meson.build

```diff
@@ -1,28 +1,70 @@
-
-# TODO: Fix the approach to vcpkg (???).
-vcpkg_target_triplet = 'x86-windows-static-md'
-vcpkg_dir = 'vcpkg_installed/' + vcpkg_target_triplet
-vcpkg_dir_rel = '../../' + vcpkg_dir
-vcpkg_dir_abs = meson.current_source_dir() + '/' + vcpkg_dir
-vcpkg_include_dir_rel = vcpkg_dir_rel + '/include/'
-vcpkg_lib_dir_debug_rel = vcpkg_dir_rel + '/debug/lib/'
-vcpkg_lib_dir_release_rel = vcpkg_dir_rel + '/lib/'
-# vcpkg_include_dir_abs = vcpkg_dir_abs + '/include/'
-vcpkg_lib_dir_debug_abs = vcpkg_dir_abs + '/debug/lib/'
-vcpkg_lib_dir_release_abs = vcpkg_dir_abs + '/lib/'
-
-include_dirs = ['.', '..', '../../lib/NWN2/include', vcpkg_include_dir_rel]
-lib_dirs = [meson.current_source_dir() + '/lib/NWN2']
-
-if get_option('buildtype') == 'debug'
-    include_dirs += vcpkg_lib_dir_debug_rel + '/mswud'
-    lib_dirs += vcpkg_lib_dir_debug_abs
+project(
+    'NWNX4',
+    ['cpp', 'c'],
+    default_options : [
+        # 'default_library=static',
+        'cpp_std=c++latest',
+        'bindir=.',
+        'libdir=.',
+        'datadir=.',
+    ],
+)
+
+# Global compiler arguments
+cppcompiler = meson.get_compiler('cpp')
+add_project_arguments('/permissive', '/DWIN32', '/D_WINDOWS', '/D_USRDLL', '/DTEST_EXPORTS', '/D_UNICODE', '/DUNICODE', language: ['cpp', 'c'])
+if get_option('buildtype') == 'release'
+    add_project_arguments('/DNDEBUG', language: ['cpp', 'c'])
+    # add_global_link_arguments('/GL', language: ['cpp', 'c'])
 else
-    include_dirs += vcpkg_lib_dir_release_rel + '/mswu'
-    lib_dirs += vcpkg_lib_dir_release_abs
+    add_project_arguments('/D_DEBUG', language: ['cpp', 'c'])
 endif
 
-subdir('src')
+# Include & lib paths
+nwnx4_repo_path = meson.global_source_root()
+include_dirs = include_directories([
+    'lib/NWN2/include',
+    'vcpkg_installed/x86-windows-static-md/include',
+])
+lib_dirs = [
+    nwnx4_repo_path + '/lib/NWN2',
+    nwnx4_repo_path + '/vcpkg_installed/x86-windows-static-md/lib',
+]
+if get_option('buildtype') == 'release'
+    lib_dirs += nwnx4_repo_path + '/vcpkg_installed/x86-windows-static-md/lib'
+else
+    lib_dirs += nwnx4_repo_path + '/vcpkg_installed/x86-windows-static-md/debug/lib'
+endif
+
+
+# Target list
+subdir('src/hook')
+subdir('src/controller')
+subdir('src/gui')
+
+nwscript_includes = []
+
+# subdir('src/plugins/chat') # Obsolete ?
+subdir('src/plugins/funcs')
+# subdir('src/plugins/leto') # Obsolete / not ported
+subdir('src/plugins/mysql')
+subdir('src/plugins/pickpocket')
+subdir('src/plugins/profiler')
+# subdir('src/plugins/ruby') # Not ported
+subdir('src/plugins/sqlite')
+subdir('src/plugins/time')
+# subdir('src/plugins/vaultster') # TODO
+subdir('src/plugins/xp_bugfix')
+subdir('src/plugins/xp_fastboot')
+subdir('src/plugins/xp_objectattributes')
+subdir('src/plugins/xp_srvadmin')
+
+
+custom_target(
+    'nwscript-includes.erf',
+    command: ['vcpkg_installed/nwn-lib-d-tools/nwn-erf', 'create', '-o', get_option('datadir') + '/nwscript-includes.erf', get_option('datadir') + '/nwscript'],
```

Comment:
vcpkg_installed packages are installed and ignored by the Git repo. If this is a package necessary, we should either install it through vcpkg.json or the lib/NWN2 folder.

Reply:
oops, this shouldn't have been committed :/
The original idea was to download nwn-lib-d-tools with the bootstrap script, then create the ERF file with meson.
But I chose instead to build the erf file only in the CI process, because it's not very useful for people wanting to build nwnx4 (meson already copies all required NSS files into the nwscript_include folder)

I'm removing the custom_target from the PR